crates/biome_html_formatter/src/lib.rs

```diff
@@ -205,10 +206,29 @@ where
 
     /// Formats the node without comments. Ignores any suppression comments.
     fn fmt_node(&self, node: &N, f: &mut HtmlFormatter) -> FormatResult<()> {
-        self.fmt_fields(node, f)?;
+        if let Some(range) = self.embedded_node_range(node) {
+            // Tokens that belong to embedded nodes are formatted later on,
+            // so we track them, even though they aren't formatted now during this pass.
+            let state = f.state_mut();
+            for token in node.syntax().tokens() {
+                state.track_token(&token);
+            }
+
+            f.write_elements(vec![
+                FormatElement::Tag(StartEmbedded(range)),
+                FormatElement::Tag(EndEmbedded),
```

Comment:
Is there ever a situation where `EndEmbedded` doesn't immediately follow `StartEmbedded`? If not, doesn't it make more sense to just use a "self-closing" `Embedded(range)` element, or is this just a convention we're following?